build_runner/lib/src/environment/io_environment.dart

```diff
@@ -32,7 +32,7 @@ class IOEnvironment implements BuildEnvironment {
   final bool _verbose;
 
   IOEnvironment(PackageGraph packageGraph, bool assumeTty, {bool verbose})
-      : _isInteractive = assumeTty ?? _canPrompt(),
+      : _isInteractive = assumeTty ? true : _canPrompt(),
```

Comment:
`assumeTty || _canPrompt()`?

Reply:
actually needed to go with `assumeTty == true || _canPrompt()` because we pass nulls for this today....